### DISCUSSION TITLE
Questions are not included in the hierarchy when adding QuestionSet to Course

### DISCUSSION THREAD

**Discussion by Debayanpaul1593** [Sunbird-Knowlg/Community#138] [Q&A] [answered]
I have a QuestionSet hierarchy say
- QuestionSet 1
   - Question 1

I need to add this to a Course Unit in a Course hierarchy. I am using the Add Hierarchy API to add the QuestionSet to the Course
```
PATCH {{content_service}}/content/v3/hierarchy/add
{
    "request": {
        "rootId": "<course-id>",
        "unitId": "<unit-id>",
        "children": [
            "<question-set-id>"
        ]
    }
}
```
However on inspecting the hierarchy I see that the QuestionSet has been added but the child Question has not been added. Is there any  particular job to handle the hierarchy update in this case?

**Comment by pallakartheekreddy** [comment, marked as answer]
@Debayanpaul1593 did you publish the Question Set? Please ensure the question set is published before adding it to any collection (Course). 
In the system currently, question is not a playable object because of that when you add any published question set under collection, questions under question set will not show in the collection hierarchy. 
@maheshkumargangula

**Comment by Debayanpaul1593** [comment]
@pallakartheekreddy thanks for the update. I am trying the same. I have come upon a configuration
`val printServiceBaseUrl: String = config.getString("print_service.base_url")`
but I am not sure which repo this is supposed to point to. It is causing exception in QuestionSet publish job because of which I am unable to publish my QS. Can the team please tell me what is print service and send me the repo link?

**Reply by Debayanpaul1593** [reply]
We have found this [Repo](https://github.com/project-sunbird/print-service). Is this the correct one?

**Reply by pallakartheekreddy** [reply]
Yes @Debayanpaul1593
